### ISSUE TITLE
NullPointerException in license-maven-plugin:1.12:add-third-party

### ISSUE THREAD

**Issue by Matze2** [mojohaus/license-maven-plugin#74]
It seems to happen when the plugin tries to resolve a non-existing third-party properties of an artifact.

```
Caused by: java.lang.NullPointerException
	at org.apache.maven.artifact.repository.layout.DefaultRepositoryLayout.pathOf(DefaultRepositoryLayout.java:48)
	at org.apache.maven.artifact.repository.MavenArtifactRepository.pathOf(MavenArtifactRepository.java:85)
	at org.codehaus.mojo.license.api.DefaultThirdPartyTool.resolveThirdPartyDescriptor(DefaultThirdPartyTool.java:764)
	at org.codehaus.mojo.license.api.DefaultThirdPartyTool.resolvThirdPartyDescriptor(DefaultThirdPartyTool.java:313)
	at org.codehaus.mojo.license.api.DefaultThirdPartyTool.loadThirdPartyDescriptorsForUnsafeMapping(DefaultThirdPartyTool.java:220)
	at org.codehaus.mojo.license.api.DefaultThirdPartyHelper.loadThirdPartyDescriptorForUnsafeMapping(DefaultThirdPartyHelper.java:165)
	at org.codehaus.mojo.license.api.DefaultThirdPartyHelper.createUnsafeMapping(DefaultThirdPartyHelper.java:241)
	at org.codehaus.mojo.license.AddThirdPartyMojo.createUnsafeMapping(AddThirdPartyMojo.java:299)
	at org.codehaus.mojo.license.AbstractAddThirdPartyMojo.init(AbstractAddThirdPartyMojo.java:406)
	at org.codehaus.mojo.license.AbstractLicenseMojo.execute(AbstractLicenseMojo.java:169)
	... 22 more
```

I debugged a little bit and in some cases resolveArtifact in DefaultThirdPartyTool:749 returns an empty file and in some cases it throws ArtifactNotFoundException.
In the latter case, the field "artifact" in ArtifactNotFoundException is null, only the single fields artifactId, groupId, version etc. are set.

The used ArtifactNotFoundException constructor in 3.3.9 does not pass the artifact itself but the single fields only:
```
    protected ArtifactNotFoundException( String message, Artifact artifact,
                                         List<ArtifactRepository> remoteRepositories, Throwable cause )
    {
        this( message, artifact.getGroupId(), artifact.getArtifactId(), artifact.getVersion(), artifact.getType(),
              artifact.getClassifier(), remoteRepositories, artifact.getDownloadUrl(), artifact.getDependencyTrail(),
              cause );
    }
```
Therefore "artifact" remains null in the exception.

Later on, this field is accessed which leads to the NPE above.

If more info is required just let me know.

Thanks,
Matthias

**Comment by Matze2** [comment]
I looked in my local repository.
There are artifact entries with an empty xyz-third-party.properties file and entries without a third-party.properties.
The latter causes the NPE.

**Comment by mrdanpsmith** [comment]
I think this is a duplicate of or related to:

https://github.com/mojohaus/license-maven-plugin/issues/70

**Comment by Matze2** [comment]
Thank you for your offer, but we already work with a patched version of 1.9.
The goal is to use an official version again.

**Comment by tjuchniewicz** [comment]
Works for me using 1.10.

**Comment by Matze2** [comment]
And? The report was for version 1.12.

**Comment by tjuchniewicz** [comment]
I wanted only to confirm that possible workaround: downgrade to 1.10 fix this issue.

**Comment by mrdanpsmith** [comment]
Yes; 1.10 works.

**Comment by tchemit** [comment]
Ok fix by #70 (thanks for the report).
